### PR TITLE
Move documentation to docs folder

### PR DESCRIPTION
**Pull request by tombye**
Mainly so changes to code can include changes to the documentation in their pull request.

### CODE REVIEW

**Comment by joelanman** [comment]
link to the old wiki:

https://github.com/tombye/express_prototype/blob/move-documentation-to-docs-folder/app/views/index.html#L31

**Comment by joelanman** [comment]
not sure when it happened but this link is broken:

https://github.com/tombye/express_prototype/blob/move-documentation-to-docs-folder/app/views/examples/index.html#L19

personally I think it should be /examples/blank

**Comment by joelanman** [comment]
all the Documentation links 404 for me:

https://github.com/tombye/express_prototype/tree/move-documentation-to-docs-folder

**Comment by tombye** [comment]
They are relative links. Think the branch name is missing because it's not going far enough up the folder path.

**Comment by joelanman** [comment]
they're not relative to the branch, they're like this:

`blob/master/docs/getting-started.md`

when they should be

`docs/getting-started.md`

I think

**Comment by tombye** [comment]
Oh, you mean those in the README? Hard to tell without some context, are there any others you were talking about?

**Comment by joelanman** [comment]
sorry yes, I linked to the page where the links appear, and where they 404. Havent come across any others

**Comment by joelanman** [comment]
seems like all the `/docs` files might need H1s

**Comment by tombye** [comment]
Good call, more semantics always useful.

File: docs/creating-routes.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+* **request** : the 1st parameter sent to the callback, an object representing the HTTP request made
+* **response** : the 2nd parameter sent to the callback, an object representing the HTTP response that will be sent
+* **response.render** : method of the response object used to create a page to send back to the browser that made the request
+* **template** : the 1st parameter sent to response.render, the name of the template file used to render the page, minus its `.html` extension
+* **data** : the 2nd parameter sent to response.render, an object containing variables to send into the template
+
+So as an example, a request for the URL `http://localhost:3000/sample` has this route:
+
+    get('/sample', function(req, res) {
+        res.render('sample', { 'assetPath' : '/public/' });
```
Comment (joelanman) [inline docs/creating-routes.md]: assetPath is no longer sent manually

I think sample should live in `/views/examples`, and we could send an example variable to it, like an H1 or something

File: docs/deploying.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+If you don't want the wider world to be looking at your prototype then you can add some basic username/password authentication. Only one set of credentials is supported at this time.
+
+The first step is set some environment variables on your Heroku instance. You can do this in the Heroku admin console (under Settings ▶ Config Variables) or by running the following shell commands:
+
+`heroku config:set NODE_ENV=production`
+`heroku config:set USERNAME=username_here`
+`heroku config:set PASSWORD=password_here`
+
+Then, at the start of app.js, authenticate against the provided username and password when running in the production environment (i.e. Heroku):
```
Comment (joelanman) [inline docs/deploying.md]: this is out of date, this code is already in the kit - only the heroku commands should be necessary

**Comment by joelanman** [comment]
general comment I just thought of - the wiki has an index of all pages, which we dont have any more... I guess we have one if you look at /docs

**Comment by joelanman** [comment]
LGTM

**Pull request merged**